### PR TITLE
update airflow docs, combine two CI jobs

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

Follow-up to https://github.com/cal-itp/data-infra/pull/2467, continuing to update infra docs before I am out next week

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation
- [ ] agencies.yml

## How has this been tested?
This PR does combine two CI jobs; the remaining job will run even if just `airflow/README.md` changes, and we can confirm that the requirements update is skipped.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,13 @@ Documentation for this codebase lives at [docs.calitp.org/data-infra](https://do
...
-* `airflow` contains the local dev setup and source code for Airflow DAGs (ie, ETLs)
-* `airflow/data/agencies.yml` contains catalogs for all transit agencies in CA's GTFS data.
-* `ci` contains continious integration and deployment scripts using GH actions.
-* `docs` builds the docs site.
-* `kubernetes` contain helm charts, scripts and more for deploying apps and such on our kubernetes cluster.
-* `script` contains associated scripts (mostly python) that are ad hoc.
-* `services` contain apps that we write and deploy to kubernetes.
+* [./airflow](./airflow) contains the local dev setup and source code for Airflow DAGs (i.e. ETL)
+* [./ci](./ci) contains continuous integration and deployment scripts using GitHub actions.
+* [./docs](./docs) builds the docs site.
```
Comment (lauriemerrell) [inline README.md:11]: List URL for what `the docs site` refers to?

File: airflow/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Airflow / ETL
+# Airflow
 
 The following folder contains the project level directory for all our apache airflow ETLs, which are deployed automatically to Google Cloud Composer from the `main` branch.
```
Comment (lauriemerrell) [inline airflow/README.md:3]: nit: `Apache Airflow` capitalized?

File: airflow/README.md [hunk 3]
```diff
@@ -34,34 +34,23 @@ gcloud init
...
 
 To access the web UI, visit `http://localhost:8080`.
 The default login and password for airflow's image are both "airflow".
 
-To run a DAG, you can either test it via the web UI or run a one-off with:
+You may execute DAGs via the web UI, or just specific individual tasks via the CLI.
 
 ```console
-docker-compose run airflow dags trigger <dag_id>
+docker-compose run airflow tasks test download_gtfs_schedule_v2 download_schedule_feeds 2022-04-01T00:00:00
```
Comment (lauriemerrell) [inline airflow/README.md:53]: Do people need access to Secrets Manager to run the downloader DAG? How does that work in local Airflow?

File: docs/airflow/dags-maintenance.md [hunk 4]
```diff
@@ -1,39 +1,70 @@
...
+| `amplitude_benefits`               | No    | This will be replaced soon by an Amplitude-to-GCS export.                                                 |
+| `create_external_tables`           | Yes   | Generally only fails due to GCP issues or definition problems; individual tasks can be cleared once fixed |
+| `deploy_dbt_docs`                  | Yes   | Manual job to deploy dbt docs to Metabase and Netlify                                                     |
+| `download_gtfs_schedule_v2`        | Yes   | Don't need to rerun more than once if multiple failures; scraped data is labeled by actual execution time |
+| `parse_and_validate_rt_v2`         | No    |                                                                                                           |
+| `payments_loader`                  | Yes   | This is deprecated, pending data consumers switching to v2 models. See below for maintenance.                                       |
+| `scrape_feed_aggregators`          | Yes   |                                                                                                           |
```
Comment (lauriemerrell) [inline docs/airflow/dags-maintenance.md:34]: Should this have the `outputs timestamped artifacts` note too?

File: docs/airflow/dags-maintenance.md [hunk 5]
```diff
@@ -1,39 +1,70 @@
...
-clear and re-run tasks as needed.
+Subject to the considerations outlined abov, backfilling can be performed by clearing historical runs in the web interface, or via the CLI:
+```shell
+gcloud composer environments run calitp-airflow-prod --location=us-west2 backfill -- --start_date 2021-04-18 --end_date 2021-11-03 -x --reset_dagruns -y -t "gtfs_schedule_history_load" -i gtfs_loader
+```
 
-(mst-payments-dags)=
 ## MST Payments - DAGs Maintenance
 
 The ETL is currently a scheduled Google Data Transfer job that transfers all files to `gcs://littlepay-data-extract-prod`
```
Comment (lauriemerrell) [inline docs/airflow/dags-maintenance.md:70]: Should this section be renamed -- is this specific to MST? Or should this be general payments? (At first I thought this was the job that stages data for direct consumption by whichever agency does that)

File: docs/datasets_and_tables/mst_payments.md [hunk 6]
```diff
@@ -17,4 +17,4 @@ The table best used for caculating ridership is `mst_ridership_materialized` tab
 
 ## DAGs Maintenance
 
-You can find further information on DAGs maintenance for MST Payments data [on this page](mst-payments-dags).
+You can find further information on DAGs maintenance for MST Payments data [on this page](dags-maintenance).
```
Comment (lauriemerrell) [inline docs/datasets_and_tables/mst_payments.md:20]: Same Q -- is this specific to MST or not?

File: docs/datasets_and_tables/overview.md [hunk 7]
```diff
@@ -12,18 +12,47 @@ kernelspec:
...
 
 | page | description | datasets |
 | ---- | ----------- | -------- |
-| [Views](./views.md) | End-user friendly data for dashboards and metrics | E.g. `views.validation_*`, `views.reports_*`, `views.gtfs_schedule_*`, `views.payments_*`, `views.gtfs_rt_*`, `views.airtable_*`, `mart_transit_database.*`, `gtfs_schedule.*` |
-| [Amplitude](./amplitude.md) | Data from Amplitude API for Benefits app | `amplitude.benefits_events `, `views.amplitude_benefits_events` |
-| [GTFS-Schedule (Deprecated)](./gtfs_schedule.md) | GTFS Schedule data for the current day | `gtfs_schedule`, `gtfs_schedule_history.*`, `gtfs_schedule_type2.*`, `gtfs_views_staging.*` |
-| [GTFS-Realtime (Deprecated)](gtfs-realtime) | GTFS Realtime data, collected every 20 seconds | `gtfs_rt.*`, `external_gtfs_rt.*`, `gtfs_rt_logs.*`, `staging.stg_rt__*` |
 | [Transit Database](./transitdatabase.md) | A representation of Cal-ITP's internal knowledge about our Transit Operators in CA and various pieces of National Transit Database statistics for ease of use | `airtable.*`, `staging.transit_database__*`, `transitstacks.*` |
 | [MST Payments](./mst_payments.md) | To-do | To-do |
```
Comment (lauriemerrell) [inline docs/datasets_and_tables/overview.md:58]: I'd remove this final MST row

File: docs/datasets_and_tables/transitdatabase.md [hunk 8]
```diff
@@ -28,9 +28,6 @@ AirTable does not currently have an effective mechanism to programmaticaly downl
 
 The data in the Airtable Transit Database is distinct but related to the following data:
 
-- [GTFS Schedule Quality Assessment Airflow pipelines](/airflow/static-schedule-pipeline) results in the [warehouse](/warehouse/overview)
-- [GTFS Schedules Datasets](gtfs_schedule)
-
 Currently neither of the above processes or datasets either rely on or contribute back to the Airflow Transit Database – this is a work in progress.  Rather, they rely on the list of transit datasets in the file [`agencies.yml`](https://github.com/cal-itp/data-infra/tree/main/airflow/data/agencies.yml) to dictate what datasets to download and assess.
```
Comment (lauriemerrell) [inline docs/datasets_and_tables/transitdatabase.md:31]: This paragraph should be deleted (re: agencies.yml)

**Review by lauriemerrell — CHANGES_REQUESTED** [review]
Thank you for tackling this, mostly just minor comments

File: docs/airflow/dags-maintenance.md (hunk 5 above)
Comment (atvaccaro) [reply docs/airflow/dags-maintenance.md:70]: Yeah it's general payments; I will rename (was trying to avoid changing since it's v1)

**Review by lauriemerrell — APPROVED** [review]
approving but would prefer that at least one of @KatrinaMKaiser and @tiffanychu90 confirm as well, ideally @charlie-costanzo too as a docs site knower

**Review by tiffanychu90 — APPROVED** [review]
definitely agree that `datasets_and_tables` pages are outdated/can be removed

**Pull request merged**